### PR TITLE
Generate missing ExpenseSheets on Service update

### PR DESCRIPTION
**Pull request by skius**

### CODE REVIEW

File: api/spec/requests/v1/services_controller_spec.rb [hunk 1]
```diff
@@ -274,6 +274,21 @@
             expect(parse_response_json(response)).to include(expected_attributes)
           end
         end
+
+        context 'when an admin user updates the ending of a service' do
```
Comment (lukasbischof) [inline api/spec/requests/v1/services_controller_spec.rb]: Maybe it is worth protecting it agains civil servant after confirmation date?

File: api/spec/services/expense_sheet_generator_spec.rb [hunk 2]
```diff
@@ -127,4 +143,17 @@
...
+
+  describe '#create_missing_expense_sheets' do
+    before do
+      expense_sheet_generator.create_expense_sheets
+      service.update ending: service.ending + 28.days
+      allow(expense_sheet_generator).to receive(:create_expense_sheets)
+    end
+
+    it 'calls create_expense_sheets with the correct arguments' do
+      expense_sheet_generator.create_missing_expense_sheets
```
Comment (lukasbischof) [inline api/spec/services/expense_sheet_generator_spec.rb]:
```suggestion
      expect { expense_sheet_generator.create_missing_expense_sheets }.to change(ExpenseSheet, :count).by(1)
```

File: api/spec/requests/v1/services_controller_spec.rb (hunk 1 above)
Comment (skius) [reply api/spec/requests/v1/services_controller_spec.rb]: Implemented this

File: api/spec/services/expense_sheet_generator_spec.rb (hunk 2 above)
Comment (skius) [reply api/spec/services/expense_sheet_generator_spec.rb]: This is already spec'd in ExpenseSheetGenerator#create_expense_sheets

**Pull request merged**